### PR TITLE
added EigenOdometry type+ conversions, added alignemte safe std container typedefs

### PR DESCRIPTION
**Pull request by markusachtelik**

### CODE REVIEW

File: mav_msgs/include/mav_msgs/conversions.h [hunk 1]
```diff
@@ -19,6 +19,10 @@
 #ifndef MAV_MSGS_CONVERSIONS_H
 #define MAV_MSGS_CONVERSIONS_H
 
+#include <Eigen/StdVector>
```
Comment (simonlynen) [inline mav_msgs/include/mav_msgs/conversions.h]: Eigen should go down into other group. https://github.com/ethz-asl/multiagent_mapping/wiki/Header-include-guide

File: mav_msgs/include/mav_msgs/conversions.h [hunk 2]
```diff
@@ -94,6 +98,30 @@ inline void eigenCommandTrajectoryPositionYawFromMsg(const CommandTrajectoryPosi
...
+  odometry->position = mav_msgs::vector3FromPointMsg(msg.pose.pose.position);
+  odometry->orientation = mav_msgs::quaternionFromMsg(msg.pose.pose.orientation);
+  odometry->velocity = mav_msgs::vector3FromMsg(msg.twist.twist.linear);
+  odometry->angular_velocity = mav_msgs::vector3FromMsg(msg.twist.twist.angular);
 }
 
+#define MAV_MSGS_CONCATENATE(x, y) x ## y
+#define MAV_MSGS_CONCATENATE2(x, y) MAV_MSGS_CONCATENATE(x, y)
+#define MAV_MSGS_MAKE_ALIGNED_CONTAINERS(eigen_type) \
+  typedef std::vector<eigen_type, Eigen::aligned_allocator<eigen_type> > MAV_MSGS_CONCATENATE2(eigen_type, Vector); \
```
Comment (simonlynen) [inline mav_msgs/include/mav_msgs/conversions.h]: EIGEN_TYPE or EigenType for consistenty

File: mav_msgs/include/mav_msgs/conversions.h [hunk 3]
```diff
@@ -94,6 +98,30 @@ inline void eigenCommandTrajectoryPositionYawFromMsg(const CommandTrajectoryPosi
...
+MAV_MSGS_MAKE_ALIGNED_CONTAINERS(EigenCommandAttitudeThrust)
+MAV_MSGS_MAKE_ALIGNED_CONTAINERS(EigenCommandMotorSpeed)
+MAV_MSGS_MAKE_ALIGNED_CONTAINERS(EigenCommandRateThrust)
+MAV_MSGS_MAKE_ALIGNED_CONTAINERS(EigenCommandTrajectoryPositionYaw)
+MAV_MSGS_MAKE_ALIGNED_CONTAINERS(EigenCommandRollPitchYawrateThrust)
+MAV_MSGS_MAKE_ALIGNED_CONTAINERS(EigenOdometry)
+
+}
+
+
```
Comment (simonlynen) [inline mav_msgs/include/mav_msgs/conversions.h]: vsp

File: mav_msgs/include/mav_msgs/conversions.h [hunk 4]
```diff
@@ -94,6 +98,30 @@ inline void eigenCommandTrajectoryPositionYawFromMsg(const CommandTrajectoryPosi
...
+  odometry->orientation = mav_msgs::quaternionFromMsg(msg.pose.pose.orientation);
+  odometry->velocity = mav_msgs::vector3FromMsg(msg.twist.twist.linear);
+  odometry->angular_velocity = mav_msgs::vector3FromMsg(msg.twist.twist.angular);
 }
 
+#define MAV_MSGS_CONCATENATE(x, y) x ## y
+#define MAV_MSGS_CONCATENATE2(x, y) MAV_MSGS_CONCATENATE(x, y)
+#define MAV_MSGS_MAKE_ALIGNED_CONTAINERS(eigen_type) \
+  typedef std::vector<eigen_type, Eigen::aligned_allocator<eigen_type> > MAV_MSGS_CONCATENATE2(eigen_type, Vector); \
+  typedef std::list<eigen_type, Eigen::aligned_allocator<eigen_type> > MAV_MSGS_CONCATENATE2(eigen_type, List); \
```
Comment (simonlynen) [inline mav_msgs/include/mav_msgs/conversions.h]: do you really want people to use list?

File: mav_msgs/include/mav_msgs/eigen_mav_msgs.h [hunk 5]
```diff
@@ -121,6 +121,28 @@ struct EigenCommandTrajectoryPositionYaw {
...
+
+  EigenOdometry(const Eigen::Vector3d& _position, const Eigen::Quaterniond& _orientation,
+                const Eigen::Vector3d& _velocity, const Eigen::Vector3d& _angular_velocity)
+      : position(_position),
+        orientation(_orientation),
+        velocity(_velocity),
+        angular_velocity(_angular_velocity) {}
+
+  EIGEN_MAKE_ALIGNED_OPERATOR_NEW
+  Eigen::Vector3d position;
```
Comment (simonlynen) [inline mav_msgs/include/mav_msgs/eigen_mav_msgs.h]: add frames of reference?

File: mav_msgs/include/mav_msgs/eigen_mav_msgs.h [hunk 6]
```diff
@@ -121,6 +121,28 @@ struct EigenCommandTrajectoryPositionYaw {
...
+  Eigen::Quaterniond orientation;
```
Comment (simonlynen) [inline mav_msgs/include/mav_msgs/eigen_mav_msgs.h]: q_GI?

File: mav_msgs/include/mav_msgs/eigen_mav_msgs.h [hunk 7]
```diff
@@ -121,6 +121,28 @@ struct EigenCommandTrajectoryPositionYaw {
...
+  Eigen::Vector3d velocity;  // Velocity in expressed in the Body frame!
```
Comment (simonlynen) [inline mav_msgs/include/mav_msgs/eigen_mav_msgs.h]: v_I

File: mav_msgs/include/mav_msgs/eigen_mav_msgs.h [hunk 8]
```diff
@@ -121,6 +121,28 @@ struct EigenCommandTrajectoryPositionYaw {
...
+  Eigen::Vector3d angular_velocity;
```
Comment (simonlynen) [inline mav_msgs/include/mav_msgs/eigen_mav_msgs.h]: w_I

**Comment by simonlynen** [comment]
lgtm

File: mav_msgs/include/mav_msgs/eigen_mav_msgs.h (hunk 7 above)
Comment (markusachtelik) [inline mav_msgs/include/mav_msgs/eigen_mav_msgs.h]: nope, let's stick with the names defined in the messages

File: mav_msgs/include/mav_msgs/conversions.h [hunk 9]
```diff
@@ -94,6 +98,30 @@ inline void eigenCommandTrajectoryPositionYawFromMsg(const CommandTrajectoryPosi
...
+inline void eigenOdometryFromMsg(const nav_msgs::Odometry& msg, EigenOdometry* odometry) {
+  assert(odometry != NULL);
+  odometry->position = mav_msgs::vector3FromPointMsg(msg.pose.pose.position);
+  odometry->orientation = mav_msgs::quaternionFromMsg(msg.pose.pose.orientation);
+  odometry->velocity = mav_msgs::vector3FromMsg(msg.twist.twist.linear);
+  odometry->angular_velocity = mav_msgs::vector3FromMsg(msg.twist.twist.angular);
 }
 
+#define MAV_MSGS_CONCATENATE(x, y) x ## y
+#define MAV_MSGS_CONCATENATE2(x, y) MAV_MSGS_CONCATENATE(x, y)
```
Comment (ffurrer) [inline mav_msgs/include/mav_msgs/conversions.h]: why can't you use `MAV_MSGS_CONCATENATE(x, y)` directly? (@simonlynen )

**Pull request merged**